jacquard/buckets/tests/test_bucket.py

```diff
@@ -18,3 +20,9 @@ def test_divisible(divisor):
 
 def test_at_least_three_buckets_per_percent():
     assert NUM_BUCKETS / 100 >= 3
+
+
+def test_can_get_empty_bucket_from_old_format():
+    session = Session({'buckets/1': []})
+    bucket = session.get(Bucket, 1)
+    assert not bucket.needs_constraints()
```

Comment:
Looks like `needs_constraints` here is being used for its side-effects. Is there no direct thing we could use? Alternatively please comment that this is here purely for the side-effect (and specify which side-effect so that it can be changed in future if needed).

Reply:
f2fd224